nossomercadolivre-api/src/main/java/com/github/nossomercadolivre/User.java

```diff
@@ -0,0 +1,140 @@
+package com.github.nossomercadolivre;
+
+import com.github.nossomercadolivre.exception.InvalidPasswordException;
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import javax.persistence.*;
+import javax.validation.constraints.*;
+import java.io.Serializable;
+import java.time.Instant;
+import java.util.Objects;
+
+import static javax.persistence.GenerationType.IDENTITY;
+
+@Entity
+@Table
+public class User implements Serializable {
+
+    @Id
+    @GeneratedValue(strategy = IDENTITY)
+    private Long id;
+
+    @CreatedDate
+    @NotNull
+    @Column(updatable = false, nullable = false)
+    private Instant createdDate = Instant.now();
+
+    @NotNull
+    @NotEmpty
+    @Email
+    @Size(min = 5, max = 100)
+    @Column(nullable = false, unique = true, length = 100)
+    private String login;
+
+    /**
+     * Password encrypted
+     */
+    @NotEmpty
+    @NotNull
+    @Size(min = 6, max = 200)
+    @Column(nullable = false, length = 200)
+    private String password;
+
+    @Deprecated
+    /**
+     * Just to frameworks
+     */
+    private User() {
+    }
+
+    public User(Instant createdDate, String login, String password) {
+        this.createdDate = createdDate;
+        this.login = login;
+        this.password(password);
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    private void setId(Long id) {
```

Comment:
*Idealmente, todo código escrito deveria ser chamado por alguém. Se não tem ninguém chamando, ele não deveria existir.*

Pq tem esse e os outros setters?

Reply:
Talvez o spring/hibernate tenha mudado, mas lembro de uma época que só funcionava com o set/get implementado por causa do modo que a injeção de dependência funcionava. Então, quando não queria que um atributo fosse modificado, deixava pistas de que não deveria utilizar o set para substituir o valor por ser readonly.

Testei aqui apagando o set privado, e tudo funcionou.
Aproveitei pra tirar o construtor padrão também, e esse ainda é obrigatório, tomei um: "No default constructor for entity:  : com.github.nossomercadolivre.User;". Então, vou remover o set privado. Se aparecer algum cenário que ele seja obrigatório, volto ele pro código.